libmscore/repeatlist.cpp

```diff
@@ -399,24 +399,29 @@ void RepeatList::unwind()
 
 Measure* RepeatList::jumpToStartRepeat(Measure* m)
       {
-      Measure* nm;
-      //
-      // go back to start of repeat or section break
-      // handle special case of end repeat (Measure m) has a section break
-      //
-      for (nm = m; nm && nm != _score->firstMeasure(); nm = nm->prevMeasure()) {
-            if (nm->repeatFlags() & Repeat::START || (nm->sectionBreak() && m != nm)) {
-                  if (nm->sectionBreak() && nm->nextMeasure())
-                        nm = nm->nextMeasure();
-                  break;
-                  }
-            }
+      // finalize the previous repeat segment
       rs->len = m->tick() + m->ticks() - rs->tick;
       append(rs);
 
+      // search backwards until find start of repeat
+      while (true) {
+
+            if (m->repeatFlags() & Repeat::START)
+                  break;
+
+            if (m == _score->firstMeasure())
```

Comment:
because I perform this test & break before the next test which dereferences m->prevMeasure()->sectionBreak(), I am ensured that m->prevMeasure() exists.  Again, the only time when such event m->prevMeasure() not being valid would be if there was some broken link in the score's measure list.